pymeasure/instruments/rigol/rigol_ds1000z.py

```diff
@@ -0,0 +1,849 @@
+    HIGH_RESOLUTION = "HRES"
+
+
+class WaveformMode(StrEnum):
+    """Waveform acquisition modes supported by the scope."""
+    NORMAL = "NORM"
+    MAXIMUM = "MAX"
+    RAW = "RAW"
+
+
+class WaveformFormat(StrEnum):
+    """Waveform data formats supported by the scope."""
+    BYTE = "BYTE"
+    WORD = "WORD"
+    ASCII = "ASC"
+
+
+_WAVEFORM_CHUNK_LIMITS: dict[WaveformFormat, int] = {
+    WaveformFormat.BYTE: 250_000,
+    WaveformFormat.WORD: 125_000,
+    WaveformFormat.ASCII: 15_625,
+}
+
+
+# Display enums
+
+class DisplayType(StrEnum):
+    VECTORS = "VECT"
+    DOTS = "DOTS"
+
+
+class DisplayGrid(StrEnum):
+    FULL = "FULL"
+    HALF = "HALF"
+    NONE = "NONE"
+
+
+class DisplayImageFormat(StrEnum):
+    BMP24 = "BMP24"
+    BMP8 = "BMP8"
+    PNG = "PNG"
+    JPEG = "JPEG"
+    TIFF = "TIFF"
+
+
+def _parse_waveform_preamble(reply: str) -> dict[str, Union[int, float]]:
+    """Parse the `:WAV:PRE?` reply into a structured dictionary."""
+
+    fields = reply.split(",")
+    if len(fields) != 10:
+        raise RuntimeError(f"Unexpected waveform preamble: {reply!r}")
+
+    fmt_code, acquisition_type, points, averages = map(int, fields[:4])
+    (x_increment, x_origin, x_reference,
+     y_increment, y_origin, y_reference) = map(float, fields[4:])
+
+    return {
+        "format_code": fmt_code,
+        "acquisition_type": acquisition_type,
+        "points": points,
+        "averages": averages,
+        "x_increment": x_increment,
+        "x_origin": x_origin,
+        "x_reference": x_reference,
+        "y_increment": y_increment,
+        "y_origin": y_origin,
+        "y_reference": y_reference,
+    }
+
+
+def _parse_tmc_block(read_bytes_func) -> bytes:
+    """Read and return the payload of a SCPI binary block (#Nxxxx...payload)."""
+    header = read_bytes_func(2)
+    if len(header) != 2 or not header.startswith(b"#"):
+        raise RuntimeError("Invalid binary block header for display data.")
+    length_digits = int(header[1:2].decode("ascii"))
+    length = int(read_bytes_func(length_digits).decode("ascii"))
+    payload = read_bytes_func(length)
+    # consume optional trailing terminator without failing if absent
+    try:
+        read_bytes_func(1, break_on_termchar=True)
+    except Exception:
+        pass
+    return payload
+
+
+def _normalize_memory_depth_value(value: Union[str, int]) -> tuple[str, Optional[int]]:
+    """Normalise memory depth input, returning the SCPI text and numeric form.
+
+    Returns
+    -------
+    tuple
+        (value_as_string, numeric_points or None if AUTO)
+    """
+
+    if isinstance(value, str):
+        candidate = value.strip().upper()
+        if candidate == "AUTO":
+            return "AUTO", None
+        try:
+            numeric = int(candidate)
+        except ValueError as exc:
+            raise ValueError(f"Unsupported memory depth: {value!r}") from exc
+    else:
+        numeric = int(value)
+        candidate = str(numeric)
```

Comment:
If `candicate` is just a string version of `numeric`, why do you need both?
Won't it be sufficient to have one?

You could use None or float("NaN") or float("inf") for auto, if you want to.